cli/flox-watchdog/src/main.rs

```diff
@@ -64,26 +65,40 @@ pub struct Cli {
     pub disable_metrics: bool,
 }
 
+fn main() -> ExitCode {
+    let args = Cli::parse();
+
+    // Initialization
+    let log_file = &args
+        .log_dir
+        .as_ref()
+        .map(|dir| dir.join(format!("watchdog.{}.log", args.pid)));
+
+    init_logger(log_file)
+        .context("failed to initialize logger")
+        .unwrap();
+    let _sentry_guard = (!args.disable_metrics).then(init_sentry);
+
+    // Main
+    match run(args) {
+        Err(_) => ExitCode::FAILURE,
+        Ok(_) => ExitCode::SUCCESS,
+    }
+}
+
 #[instrument("watchdog",
+    err(Debug),
```

Comment:
(non-blocking observation)

> The run method is annotated with tracing::instrument(err) after tracing is initialised. This ensures traces are recorded and errors are attributed to the watchdog span.

Neat, this will make it easier to identify this kind of issue:

- https://github.com/flox/flox/issues/2002#issuecomment-2316983693

It now appears in log:

```
(2) ~/projects/flox/flox on HEAD (f3c8c0f8) [$!+]
% _FLOX_WATCHDOG_LOG_LEVEL=debug flox-watchdog --registry ~/.local/share/flox/env-registry.json --hash ee4e7845 --socket /dev/null --log-dir ~/tmp --pid 12345 2>/dev/null || cat ~/tmp/watchdog.12345.log 
2024-09-05T10:13:32.339493Z DEBUG flox_watchdog::sentry: No Sentry DSN set, skipping Sentry initialization
2024-09-05T10:13:32.339676Z DEBUG watchdog{pid=12345 registry="/Users/dcarley/.local/share/flox/env-registry.json" dot_flox_hash="ee4e7845" socket="/dev/null" log_dir="/Users/dcarley/tmp"}: flox_watchdog: starting
2024-09-05T10:13:32.339730Z ERROR watchdog{pid=12345 registry="/Users/dcarley/.local/share/flox/env-registry.json" dot_flox_hash="ee4e7845" socket="/dev/null" log_dir="/Users/dcarley/tmp"}: flox_watchdog: error=failed to register watcher

Caused by:
    No such process (os error 3)
```